limbus/core/component.py

```diff
@@ -1,24 +1,31 @@
 """Component definition."""
 from __future__ import annotations
 from abc import abstractmethod
-from typing import List, Optional, TYPE_CHECKING, Callable
+from typing import List, Optional, TYPE_CHECKING, Callable, Type
 import logging
 import asyncio
 import traceback
 import functools
 
-import torch.nn as nn
+try:
+    import torch.nn as nn
+    base_cls: Type = nn.Module
+except ImportError:
+    base_cls = object
 
 from limbus.core.params import Params, InputParams, OutputParams
 from limbus.core.states import ComponentState, ComponentStoppedError
 # Note that Pipeline class cannot be imported to avoid circular dependencies.
 if TYPE_CHECKING:
     from limbus.core.pipeline import Pipeline
 
-
 log = logging.getLogger(__name__)
 
 
+class Module(base_cls):  # noqa: D101
```

Comment:
add class definition inside the try except to make the code a bit cleaner

Reply:
mypy complains, cannot be changed.

Reply:
to me there's no point to have runtime polymorphism here. A component should be as much generic as possible e.g object. If for some reason a user wants to inherit properties from pytorch, he can easily create a subclass TorchComponent. But for the use cases i'm seeing how we use limbus now i don't see why could be useful to have a component that is a nn.Module other than serializing. Even with that, you can subclass and implement specific functionality. Other concerns are when you have installed torch for running a model, but you really want to keep the component as light as possible, then you are forced to inherit from torch Module. This pattern is not scalable and increases the api complexity as soon as you increase the numbers of dependencies.

Reply:
I partially disagree but I think I have a solution XD. 

I do not want people creating classes for things we (maybe only I) want to support by default (torch and numpy) so the only solution (salomonic one) I see is creating that TorchComponent  in limbus.core and allow people to use Torch or not. If someday we want to add JAX or others we will only need to add a new class. (This will require a another minor refactor in several repos)

Anyway remember that limbus was created to support kornia and torch and we are drifting at least from my mental roadmap (not a problem if this proposed changes do not affect it, but basically I expect somebody else taking care of parts of the code that drift from the original idea, basically I cannot maintain 10 frameworks and make all of them compatible, and YES for me compatibility is a must taking into consideration all the repos I am developing around limbus).